interview-questions.md

```diff
@@ -7,60 +7,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Researched answer:
+  Researched answer: Git is a developer tool used to deal with projects ranging from small to very large. Git can be used to track changes made in different branches of code and allow multiple people to work on the same projects together. The difference between Git and Github is, git is a control system that will allow you to track of your branches you are working on while github is the holder of all your code and creates save points every time you upload to your cloud. You will be able to visit these past points at any given time to check on your work.
 
 
 
 2. What types of evaluations will return a Boolean value?
 
-  Your answer:
+  Your answer: asking 2 values are equal will return a Booleon value.
 
-  Researched answer:
+  Researched answer: By using if/else if conditions you can true or false depending on the outcome.
 
 
 
 3. What is an index? What is the difference between index and value?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: An index is only 1 character of what is displayed while a value is all of what is being displayed.
 
+  Researched answer: An index is a character that is specifically called upon in a string or array. The difference between a value and an index is a value is used to define properties, like, defining the string or number of a variable while an index is a character that is specifically called on within a string.
 
 
 4. What is a built-in method?
 
-  Your answer:
+  Your answer: Built-in methods are tools a program has to help the user complete different tasks quickly
 
-  Researched answer:
+  Researched answer: Built-in methods are functions coded into web pages to do tasks for the user efficiently.
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: A very helpful way to pair program is to have the navigator be ready with a plan and have all the tools ready before starting. There are 2 roles to paired programming. The Driver and The Navigator. The navigator is in control of having a plan/map for the project and tells the driver what to code as well as watching closely for any typos the driver may have made. The driver's duties consist of actually typing the code based off of what the navigator is telling him,
 
-  Researched answer:
+  Researched answer: In paired programming you will have two people working on the same code as a driver and a navigator. These two will switch roles of mapping and watching one another and typing the code. The driver role is in control of the computer and typing all the code in based off ideas given. The Navigator is in charge of watching everything the driver types for typos and plan out a map on which way to take the project.
```

Comment:
Very Good! you might already have this in mind: remember, just like with driving a car-- you don't want a back seat driver telling you when to hit the pedal and brakes-- you want a navigator who is doing research, finding the best routes and generally giving info about whats coming up; Likewise, A navigator should do everything you said but not tell the driver what to type or how to type it- just what the code should accomplish. The driver is free to choose how to get that done.